{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[],"authorship_tag":"ABX9TyPeG4HeGvv+8VP+fW1zq3S/"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"markdown","source":["## What you need to do before this tutorial?\n","\n","(1) Watch this Youtube video:\n","\n","60 Linux Commands you NEED to know (in 10 minutes)\n","\n","https://www.youtube.com/watch?v=gd7BXuUQ91w\n","\n","(2) Read this cheat sheet\n","\n","Cheat Sheet for Linux Command\n","\n","https://www.guru99.com/linux-commands-cheat-sheet.html\n"],"metadata":{"id":"xy4IZn3ikkT-"}},{"cell_type":"markdown","source":["**BIO/CSE/MBI 466/566  Tutorial 01  Linux Commands**\n","\n","**1. Introduction**\n","\n","Linux Operating System (OS) is one of three popular OS systems: Windows, Mac OS and Unix. Linux is a personalized version of Unix.  Linux is the major OS for bioinformatics.\n","\n","In this tutorial we will learn Linux and its basic commands.\n","\n","For this course, you will use Linux in two platform:\n","* (A)   Colab in your University Gmail accounts\n","* (B)   Miami University RedHawk cluster (You will need it for your final project)\n","* (C)   Liang Lab 10.34.229.118 Web Server (You will need it for your final project)\n","\n","The difference in terms of Linux commands is that for (A), you need to add \"!\" symbol before any of your Linux commands whereas you do not need to add \"!\" for (B) and (C)."],"metadata":{"id":"c0IGoH_PrYB4"}},{"cell_type":"markdown","source":["**2. Basic Linux Commands and Shell**\n","\n","* Learn the directory and file structure in Linux.\n","* Get yourself familiar with basic Linux commands (pwd, ls, cd, cat, echo).\n","* Understand the concepts of Shell and bash shell."],"metadata":{"id":"4kFZosYf9Bx-"}},{"cell_type":"code","source":["! pwd       # Check your current folder"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"I0Zfo66tsrlw","executionInfo":{"status":"ok","timestamp":1751908892038,"user_tz":240,"elapsed":110,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"b54f7ad7-5af9-46a6-ca1d-24b92a5a57b3"},"execution_count":16,"outputs":[{"output_type":"stream","name":"stdout","text":["/content\n"]}]},{"cell_type":"code","source":["! ls -al    # List the content of your current folder"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"ksxnSajIsnp4","executionInfo":{"status":"ok","timestamp":1751908893576,"user_tz":240,"elapsed":103,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"a7977dc7-8784-4add-d97a-5d5e7a9ffa74"},"execution_count":17,"outputs":[{"output_type":"stream","name":"stdout","text":["total 20\n","drwxr-xr-x 1 root root 4096 Jul  7 17:00 .\n","drwxr-xr-x 1 root root 4096 Jul  7 16:33 ..\n","drwxr-xr-x 4 root root 4096 Jul  1 21:04 .config\n","drwxr-xr-x 2 root root 4096 Jul  7 17:00 genome\n","drwxr-xr-x 1 root root 4096 Jul  1 21:04 sample_data\n"]}]},{"cell_type":"code","source":["! ls /      # List the content of your root folder for you cloud-based linux machine\n","           # Make sure you know how to differentiate a subfolder and a file"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"TMq6Vvolwsj1","executionInfo":{"status":"ok","timestamp":1751908895000,"user_tz":240,"elapsed":113,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"1987ce4c-bdb7-4ce9-d4b3-e40b7d388f32"},"execution_count":18,"outputs":[{"output_type":"stream","name":"stdout","text":["bin\t\t\t    kaggle\t\t      opt\t\t sys\n","boot\t\t\t    lib\t\t\t      proc\t\t tmp\n","content\t\t\t    lib32\t\t      python-apt\t tools\n","cuda-keyring_1.1-1_all.deb  lib64\t\t      python-apt.tar.xz  usr\n","datalab\t\t\t    libx32\t\t      root\t\t var\n","dev\t\t\t    media\t\t      run\n","etc\t\t\t    mnt\t\t\t      sbin\n","home\t\t\t    NGC-DL-CONTAINER-LICENSE  srv\n"]}]},{"cell_type":"code","source":["! ls /etc/  # List the content of the folder of \"etc\""],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"IxqB5o1TsxuJ","executionInfo":{"status":"ok","timestamp":1751908899789,"user_tz":240,"elapsed":113,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"d51ab4de-9f79-43fc-c07e-bf10798138e1"},"execution_count":19,"outputs":[{"output_type":"stream","name":"stdout","text":["adduser.conf\t\t       java-11-openjdk\tprofile\n","alternatives\t\t       jupyter\t\tprofile.d\n","apparmor.d\t\t       kernel\t\tpulse\n","apt\t\t\t       ld.so.cache\tpython3\n","bash.bashrc\t\t       ld.so.conf\tpython3.10\n","bash_completion.d\t       ld.so.conf.d\tpython3.11\n","bindresvport.blacklist\t       legal\t\tR\n","binfmt.d\t\t       libaudit.conf\trc0.d\n","ca-certificates\t\t       libibverbs.d\trc1.d\n","ca-certificates.conf\t       libnl-3\t\trc2.d\n","ca-certificates.conf.dpkg-old  libpaper.d\trc3.d\n","cloud\t\t\t       locale.alias\trc4.d\n","cron.d\t\t\t       locale.gen\trc5.d\n","cron.daily\t\t       localtime\trc6.d\n","cron.weekly\t\t       logcheck\t\trcS.d\n","cufile.json\t\t       login.defs\trequirements3.txt\n","dbus-1\t\t\t       logrotate.d\trequirements.core.in\n","dconf\t\t\t       lsb-release\trequirements.in\n","debconf.conf\t\t       machine-id\trequirements.user.in\n","debian_version\t\t       magic\t\tresolv.conf\n","default\t\t\t       magic.mime\trmt\n","deluser.conf\t\t       mailcap\t\tsecurity\n","depmod.d\t\t       mailcap.order\tselinux\n","dpkg\t\t\t       manpath.config\tsensors3.conf\n","e2scrub.conf\t\t       mime.types\tsensors.d\n","emacs\t\t\t       mke2fs.conf\tshadow\n","environment\t\t       modprobe.d\tshadow-\n","fonts\t\t\t       modules\t\tshells\n","fstab\t\t\t       modules-load.d\tskel\n","fuse.conf\t\t       mtab\t\tssh\n","gai.conf\t\t       mysql\t\tssl\n","gitconfig\t\t       netconfig\tsubgid\n","glvnd\t\t\t       networks\t\tsubuid\n","groff\t\t\t       nsswitch.conf\tsudo.conf\n","group\t\t\t       ODBCDataSources\tsudoers\n","group-\t\t\t       odbc.ini\t\tsudoers.d\n","gshadow\t\t\t       odbcinst.ini\tsudo_logsrvd.conf\n","gshadow-\t\t       openal\t\tsysctl.conf\n","gss\t\t\t       OpenCL\t\tsysctl.d\n","gtk-3.0\t\t\t       openmpi\t\tsystemd\n","host.conf\t\t       opt\t\tterminfo\n","hostname\t\t       os-release\ttimezone\n","hosts\t\t\t       PackageKit\ttmpfiles.d\n","hosts.allow\t\t       pam.conf\t\tucf.conf\n","hosts.deny\t\t       pam.d\t\tupdate-motd.d\n","init.d\t\t\t       papersize\tvdpau_wrapper.cfg\n","inputrc\t\t\t       passwd\t\tvim\n","iproute2\t\t       passwd-\t\twgetrc\n","ipython\t\t\t       perl\t\tX11\n","issue\t\t\t       pip.conf\t\txattr.conf\n","issue.net\t\t       polkit-1\t\txdg\n"]}]},{"cell_type":"code","execution_count":20,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"p52cNOmmkjSY","executionInfo":{"status":"ok","timestamp":1751908903578,"user_tz":240,"elapsed":109,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"da6c3b89-29da-49a5-cc33-aed3e8a96a4f"},"outputs":[{"output_type":"stream","name":"stdout","text":["PRETTY_NAME=\"Ubuntu 22.04.4 LTS\"\n","NAME=\"Ubuntu\"\n","VERSION_ID=\"22.04\"\n","VERSION=\"22.04.4 LTS (Jammy Jellyfish)\"\n","VERSION_CODENAME=jammy\n","ID=ubuntu\n","ID_LIKE=debian\n","HOME_URL=\"https://www.ubuntu.com/\"\n","SUPPORT_URL=\"https://help.ubuntu.com/\"\n","BUG_REPORT_URL=\"https://bugs.launchpad.net/ubuntu/\"\n","PRIVACY_POLICY_URL=\"https://www.ubuntu.com/legal/terms-and-policies/privacy-policy\"\n","UBUNTU_CODENAME=jammy\n"]}],"source":["#####  You can use the command to check what OS runing for your Linux machine in cloud.\n","\n","! cat /etc/os-release"]},{"cell_type":"markdown","source":["####  **Task01**:  Use the graphic interface to locate this file named \"os-release\" and open it. (Hint: click the file icon on the left panel and navigate the directory and file systems."],"metadata":{"id":"PSw8yuepxcIe"}},{"cell_type":"code","source":["##### Create a folder named \"genome\", which needs to be within /content/ folder\n","! mkdir /content/genome"],"metadata":{"id":"_mXFAPs50KOP","executionInfo":{"status":"ok","timestamp":1751907635000,"user_tz":240,"elapsed":101,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}}},"execution_count":9,"outputs":[]},{"cell_type":"code","source":["! cd /content/genome"],"metadata":{"id":"_qmSFoQZ0y4Y","executionInfo":{"status":"ok","timestamp":1751907672294,"user_tz":240,"elapsed":104,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}}},"execution_count":11,"outputs":[]},{"cell_type":"code","source":["! pwd"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"qjZJc5oN1bt4","executionInfo":{"status":"ok","timestamp":1751907692546,"user_tz":240,"elapsed":105,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"831feab4-00b3-4dfc-d1ec-381e0a7ce759"},"execution_count":12,"outputs":[{"output_type":"stream","name":"stdout","text":["/content\n"]}]},{"cell_type":"code","source":["! cd /content/genome && pwd"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"EDh1hED02rCc","executionInfo":{"status":"ok","timestamp":1751908018058,"user_tz":240,"elapsed":111,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"54499c6a-a580-451b-ce57-ee2cac079227"},"execution_count":13,"outputs":[{"output_type":"stream","name":"stdout","text":["/content/genome\n"]}]},{"cell_type":"code","source":["! pwd"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"Y6E_tahC2zG_","executionInfo":{"status":"ok","timestamp":1751908042389,"user_tz":240,"elapsed":107,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"faea41e0-65e8-4f04-8aa9-3579a69cbe4b"},"execution_count":14,"outputs":[{"output_type":"stream","name":"stdout","text":["/content\n"]}]},{"cell_type":"markdown","source":["##### The above behavior sounds wired.  Why?\n","##### As we know, in Linux, a shell, which is a command-line interpreter, will take your commands and translates them into instructions that OS understand. In Colab (A), each time when \"!\" is used to issue a new command, a new, different shell will be used to interpreters and translates your command. Meanwhile, for (B) and (C), when you open a terminal, you start a shell, it will terminate until you close your terminal.\n"],"metadata":{"id":"ungSwiNd6S5O"}},{"cell_type":"code","source":["#####  How to check if Colab has the popular shell to interpreters and translates your commands?\n","! echo $SHELL"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"5KBr8hwI50GV","executionInfo":{"status":"ok","timestamp":1751909590656,"user_tz":240,"elapsed":115,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"1c925c0a-76e3-4eaf-a86a-46cd6ba301e5"},"execution_count":21,"outputs":[{"output_type":"stream","name":"stdout","text":["/bin/bash\n"]}]},{"cell_type":"markdown","source":["####  **Task02**:  Within the folder /content/genome, create a subfolder named \"human\".  Use the same shell to check the content of this newly created folder. What do you find?  What do they mean?"],"metadata":{"id":"1C9TUA46-Zzf"}},{"cell_type":"code","source":["! mkdir /content/genome/human && ls -al /content/genome/human"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"KbgTF7Ht-xlx","executionInfo":{"status":"ok","timestamp":1751910890216,"user_tz":240,"elapsed":107,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"5ccb1d20-be0c-45ea-8897-6774fe47e398"},"execution_count":22,"outputs":[{"output_type":"stream","name":"stdout","text":["total 8\n","drwxr-xr-x 2 root root 4096 Jul  7 17:54 .\n","drwxr-xr-x 3 root root 4096 Jul  7 17:54 ..\n"]}]},{"cell_type":"markdown","source":["**3. Download human genomics data**\n","\n","* Human genome sequence data (https://ftp.ensembl.org/pub/release-114/fasta/homo_sapiens/dna/)\n","* Human gene annotation GFF3 data (https://ftp.ensembl.org/pub/release-114/gff3/homo_sapiens/)\n","* Human gene annotation GTF data (https://ftp.ensembl.org/pub/release-114/gtf/homo_sapiens/)\n","* Human cDNA data (https://ftp.ensembl.org/pub/release-114/fasta/homo_sapiens/cdna/)\n","* Human ncRNA data (https://ftp.ensembl.org/pub/release-114/fasta/homo_sapiens/ncrna/)\n","* Learn other Linux commands (less, rm, cp, mv, chmod, head, tail)"],"metadata":{"id":"H7faHujE0zgk"}},{"cell_type":"code","source":["! wget https://ftp.ensembl.org/pub/release-114/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz -P /content/genome/human\n","! wget https://ftp.ensembl.org/pub/release-114/gff3/homo_sapiens/Homo_sapiens.GRCh38.114.chromosome.1.gff3.gz -P /content/genome/human\n"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"9lLxwt1qFaaa","executionInfo":{"status":"ok","timestamp":1751917979310,"user_tz":240,"elapsed":3641,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"19d67664-c636-4c8c-8c26-078c3dd000b3"},"execution_count":54,"outputs":[{"output_type":"stream","name":"stdout","text":["--2025-07-07 19:52:55--  https://ftp.ensembl.org/pub/release-114/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz\n","Resolving ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.169\n","Connecting to ftp.ensembl.org (ftp.ensembl.org)|193.62.193.169|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 69273468 (66M) [application/x-gzip]\n","Saving to: ‘/content/genome/human/Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz’\n","\n","Homo_sapiens.GRCh38 100%[===================>]  66.06M  31.8MB/s    in 2.1s    \n","\n","2025-07-07 19:52:58 (31.8 MB/s) - ‘/content/genome/human/Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz’ saved [69273468/69273468]\n","\n","--2025-07-07 19:52:58--  https://ftp.ensembl.org/pub/release-114/gff3/homo_sapiens/Homo_sapiens.GRCh38.114.chromosome.1.gff3.gz\n","Resolving ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.169\n","Connecting to ftp.ensembl.org (ftp.ensembl.org)|193.62.193.169|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 5203336 (5.0M) [application/x-gzip]\n","Saving to: ‘/content/genome/human/Homo_sapiens.GRCh38.114.chromosome.1.gff3.gz’\n","\n","Homo_sapiens.GRCh38 100%[===================>]   4.96M  7.72MB/s    in 0.6s    \n","\n","2025-07-07 19:52:59 (7.72 MB/s) - ‘/content/genome/human/Homo_sapiens.GRCh38.114.chromosome.1.gff3.gz’ saved [5203336/5203336]\n","\n"]}]},{"cell_type":"code","source":["! ls -al /content/genome/human"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"jWMQj3IxcvyB","executionInfo":{"status":"ok","timestamp":1751917998695,"user_tz":240,"elapsed":105,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"61cfae52-5602-4e8d-ec20-7481e10ddcf9"},"execution_count":55,"outputs":[{"output_type":"stream","name":"stdout","text":["total 72744\n","drwxr-xr-x 2 root root     4096 Jul  7 19:52 .\n","drwxr-xr-x 3 root root     4096 Jul  7 19:52 ..\n","-rw-r--r-- 1 root root  5203336 Feb 15 15:49 Homo_sapiens.GRCh38.114.chromosome.1.gff3.gz\n","-rw-r--r-- 1 root root 69273468 Jan 30 20:44 Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz\n"]}]},{"cell_type":"code","source":["! chmod a+rwx /content/genome/human/Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz\n","! chmod a+rwx /content/genome/human/Homo_sapiens.GRCh38.114.chromosome.1.gff3.gz\n","! ls -al /content/genome/human"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"RMdoVE4uc0bp","executionInfo":{"status":"ok","timestamp":1751918025645,"user_tz":240,"elapsed":317,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"51de93d9-08d2-4d95-d2b6-f623bf72df01"},"execution_count":56,"outputs":[{"output_type":"stream","name":"stdout","text":["total 72744\n","drwxr-xr-x 2 root root     4096 Jul  7 19:52 .\n","drwxr-xr-x 3 root root     4096 Jul  7 19:52 ..\n","-rwxrwxrwx 1 root root  5203336 Feb 15 15:49 Homo_sapiens.GRCh38.114.chromosome.1.gff3.gz\n","-rwxrwxrwx 1 root root 69273468 Jan 30 20:44 Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz\n"]}]},{"cell_type":"code","source":["##### Unzip the downloaded files. The unzipped files are in the same folder as zipped files.\n","!gunzip -k /content/genome/human/Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz\n","!gunzip -k /content/genome/human/Homo_sapiens.GRCh38.114.chromosome.1.gff3.gz"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"qVKOIiHIF9F1","executionInfo":{"status":"ok","timestamp":1751918088972,"user_tz":240,"elapsed":14347,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"c2dba6e1-44e7-431f-f4a3-6fa6455e736c"},"execution_count":58,"outputs":[{"output_type":"stream","name":"stdout","text":["gzip: /content/genome/human/Homo_sapiens.GRCh38.dna.chromosome.1.fa already exists; do you wish to overwrite (y or n)? ^C\n","gzip: /content/genome/human/Homo_sapiens.GRCh38.114.chromosome.1.gff3 already exists; do you wish to overwrite (y or n)? ^C\n"]}]},{"cell_type":"code","source":["##### Try the following commands and explain what is role of the * symbol\n","#! rm /content/genome/human/Homo_sapiens.GRCh38.114.chromosome.*.gz\n","! rm /content/genome/human/Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz\n","! rm /content/genome/human/Homo_sapiens.GRCh38.114.chromosome.1.gff3.gz"],"metadata":{"id":"5oQk3x8cH3Ce","executionInfo":{"status":"ok","timestamp":1751913011924,"user_tz":240,"elapsed":204,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}}},"execution_count":27,"outputs":[]},{"cell_type":"markdown","source":["####  **Task03**:  Understand the difference between cat and less command, using /content/genome/human/Homo_sapiens.GRCh38.dna.chromosome.1.fa as an example?"],"metadata":{"id":"aicCk_oPTwpY"}},{"cell_type":"code","source":["! less /content/genome/human/Homo_sapiens.GRCh38.dna.chromosome.1.fa\n","! cat /content/genome/human/Homo_sapiens.GRCh38.dna.chromosome.1.fa"],"metadata":{"id":"qUmol1ZdSPZE"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["! head -5 /content/genome/human/Homo_sapiens.GRCh38.dna.chromosome.1.fa"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"6i7OvxCUVQjl","executionInfo":{"status":"ok","timestamp":1751916055105,"user_tz":240,"elapsed":101,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"3a1208e7-6d15-40a9-f863-f567fafb1039"},"execution_count":30,"outputs":[{"output_type":"stream","name":"stdout","text":[">1 dna:chromosome chromosome:GRCh38:1:1:248956422:1 REF\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n"]}]},{"cell_type":"code","source":["! head /content/genome/human/Homo_sapiens.GRCh38.dna.chromosome.1.fa"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"jIB67QbNVkta","executionInfo":{"status":"ok","timestamp":1751916118082,"user_tz":240,"elapsed":106,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"51cf3b55-30f1-4f4e-92ad-df280cafd375"},"execution_count":32,"outputs":[{"output_type":"stream","name":"stdout","text":[">1 dna:chromosome chromosome:GRCh38:1:1:248956422:1 REF\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n"]}]},{"cell_type":"code","source":["! tail -5 /content/genome/human/Homo_sapiens.GRCh38.dna.chromosome.1.fa"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"xw6mKhXHVhf1","executionInfo":{"status":"ok","timestamp":1751916097217,"user_tz":240,"elapsed":108,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"2e1a14ae-9b9c-4097-fb35-95ab5bf09fec"},"execution_count":31,"outputs":[{"output_type":"stream","name":"stdout","text":["NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n"]}]},{"cell_type":"code","source":["! tail /content/genome/human/Homo_sapiens.GRCh38.dna.chromosome.1.fa"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"LhxYx5NTVpce","executionInfo":{"status":"ok","timestamp":1751916131010,"user_tz":240,"elapsed":121,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"6ed1edbf-aefc-4764-bfce-848f09a77fa2"},"execution_count":33,"outputs":[{"output_type":"stream","name":"stdout","text":["NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n","NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n"]}]},{"cell_type":"code","source":["#### How to read the manual for a given command?\n","! man head"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"0J6Q6x24V7Ho","executionInfo":{"status":"ok","timestamp":1751916202816,"user_tz":240,"elapsed":107,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"88d3f81b-e92a-4607-cd6e-9de509b0d1a1"},"execution_count":34,"outputs":[{"output_type":"stream","name":"stdout","text":["This system has been minimized by removing packages and content that are\n","not required on a system that users do not log into.\n","\n","To restore this content, including manpages, you can run the 'unminimize'\n","command. You will still need to ensure the 'man-db' package is installed.\n"]}]},{"cell_type":"code","source":["# First, update your package list\n","!sudo apt-get update\n","\n","# Then, run unminimize. Use 'yes |' to automatically confirm prompts\n","!yes | sudo /usr/local/sbin/unminimize\n","\n","# Finally, ensure man-db is installed (it might be installed by unminimize, but it's good to be sure)\n","!sudo apt-get install -y man-db"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"collapsed":true,"id":"pqTCSwVIWo-5","executionInfo":{"status":"ok","timestamp":1751916396784,"user_tz":240,"elapsed":9207,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"26032bef-3e09-4ad8-f5b9-8576d34f5642"},"execution_count":35,"outputs":[{"output_type":"stream","name":"stdout","text":["\r0% [Working]\r            \rGet:1 http://security.ubuntu.com/ubuntu jammy-security InRelease [129 kB]\n","\r0% [Connecting to archive.ubuntu.com (185.125.190.83)] [1 InRelease 12.7 kB/129\r0% [Connecting to archive.ubuntu.com (185.125.190.83)] [Waiting for headers] [C\r                                                                               \rGet:2 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease [3,632 B]\n","Get:3 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease [1,581 B]\n","Get:4 https://r2u.stat.illinois.edu/ubuntu jammy InRelease [6,555 B]\n","Hit:5 http://archive.ubuntu.com/ubuntu jammy InRelease\n","Get:6 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [128 kB]\n","Get:7 http://security.ubuntu.com/ubuntu jammy-security/multiverse amd64 Packages [48.5 kB]\n","Get:8 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [1,262 kB]\n","Get:9 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [4,763 kB]\n","Hit:10 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease\n","Get:11 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [3,092 kB]\n","Get:12 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease [24.3 kB]\n","Get:13 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  Packages [1,804 kB]\n","Get:14 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [127 kB]\n","Hit:15 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n","Get:16 https://r2u.stat.illinois.edu/ubuntu jammy/main amd64 Packages [2,752 kB]\n","Get:17 https://r2u.stat.illinois.edu/ubuntu jammy/main all Packages [9,078 kB]\n","Get:18 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [3,404 kB]\n","Get:19 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy/main amd64 Packages [51.0 kB]\n","Get:20 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1,566 kB]\n","Get:21 http://archive.ubuntu.com/ubuntu jammy-updates/multiverse amd64 Packages [56.4 kB]\n","Get:22 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [4,917 kB]\n","Fetched 33.2 MB in 3s (11.5 MB/s)\n","Reading package lists... Done\n","W: Skipping acquire of configured file 'main/source/Sources' as repository 'https://r2u.stat.illinois.edu/ubuntu jammy InRelease' does not seem to provide it (sources.list entry misspelt?)\n","This system has been minimized by removing packages and content that are\n","not required on a system that users do not log into.\n","\n","This script restores content and packages that are found on a default\n","Ubuntu server system in order to make this system more suitable for\n","interactive use.\n","\n","Reinstallation of packages may fail due to changes to the system\n","configuration, the presence of third-party packages, or for other\n","reasons.\n","\n","This operation may take some time.\n","\n","\n","Reading package lists... Done\n","Building dependency tree... Done\n","Reading state information... Done\n","man-db is already the newest version (2.10.2-1).\n","0 upgraded, 0 newly installed, 0 to remove and 52 not upgraded.\n"]}]},{"cell_type":"code","source":["! man head"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"collapsed":true,"id":"DGoFD-5UWu_T","executionInfo":{"status":"ok","timestamp":1751916413237,"user_tz":240,"elapsed":227,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"6b076378-9907-44a2-e70c-331295a01a48"},"execution_count":36,"outputs":[{"output_type":"stream","name":"stdout","text":["HEAD(1POSIX)               POSIX Programmer's Manual              HEAD(1POSIX)\n","\n","P\bPR\bRO\bOL\bLO\bOG\bG\n","       This  manual  page is part of the POSIX Programmer's Manual.  The Linux\n","       implementation of this interface may differ (consult the  corresponding\n","       Linux  manual page for details of Linux behavior), or the interface may\n","       not be implemented on Linux.\n","\n","N\bNA\bAM\bME\bE\n","       head — copy the first part of files\n","\n","S\bSY\bYN\bNO\bOP\bPS\bSI\bIS\bS\n","       head [\b[-n _\bn_\bu_\bm_\bb_\be_\br]\b] [\b[_\bf_\bi_\bl_\be...]\b]\n","\n","D\bDE\bES\bSC\bCR\bRI\bIP\bPT\bTI\bIO\bON\bN\n","       The _\bh_\be_\ba_\bd utility shall copy its input files  to  the  standard  output,\n","       ending the output for each file at a designated point.\n","\n","       Copying  shall  end at the point in each input file indicated by the -\b-n\bn\n","       _\bn_\bu_\bm_\bb_\be_\br option. The option-argument _\bn_\bu_\bm_\bb_\be_\br shall be counted in units  of\n","       lines.\n","\n","O\bOP\bPT\bTI\bIO\bON\bNS\bS\n","       The  _\bh_\be_\ba_\bd  utility  shall  conform  to  the  Base Definitions volume of\n","       POSIX.1‐2017, _\bS_\be_\bc_\bt_\bi_\bo_\bn _\b1_\b2_\b._\b2, _\bU_\bt_\bi_\bl_\bi_\bt_\by _\bS_\by_\bn_\bt_\ba_\bx _\bG_\bu_\bi_\bd_\be_\bl_\bi_\bn_\be_\bs.\n","\n","       The following option shall be supported:\n","\n","       -\b-n\bn _\bn_\bu_\bm_\bb_\be_\br The first _\bn_\bu_\bm_\bb_\be_\br lines of each input file shall be copied  to\n","                 standard output. The application shall ensure that the _\bn_\bu_\bm_\bb_\be_\br\n","                 option-argument is a positive decimal integer.\n","\n","       When a file contains less than _\bn_\bu_\bm_\bb_\be_\br lines,  it  shall  be  copied  to\n","       standard output in its entirety. This shall not be an error.\n","\n","       If no options are specified, _\bh_\be_\ba_\bd shall act as if -\b-n\bn 1\b10\b0 had been speci‐\n","       fied.\n","\n","O\bOP\bPE\bER\bRA\bAN\bND\bDS\bS\n","       The following operand shall be supported:\n","\n","       _\bf_\bi_\bl_\be      A pathname of an input file. If no _\bf_\bi_\bl_\be operands  are  speci‐\n","                 fied, the standard input shall be used.\n","\n","S\bST\bTD\bDI\bIN\bN\n","       The standard input shall be used if no _\bf_\bi_\bl_\be operands are specified, and\n","       shall be used if a _\bf_\bi_\bl_\be operand is '\b'-\b-'\b' and  the  implementation  treats\n","       the '\b'-\b-'\b' as meaning standard input.  Otherwise, the standard input shall\n","       not be used.  See the INPUT FILES section.\n","\n","I\bIN\bNP\bPU\bUT\bT F\bFI\bIL\bLE\bES\bS\n","       Input files shall be text files, but the line length is not  restricted\n","       to {LINE_MAX} bytes.\n","\n","E\bEN\bNV\bVI\bIR\bRO\bON\bNM\bME\bEN\bNT\bT V\bVA\bAR\bRI\bIA\bAB\bBL\bLE\bES\bS\n","       The following environment variables shall affect the execution of _\bh_\be_\ba_\bd:\n","\n","       _\bL_\bA_\bN_\bG      Provide  a  default  value for the internationalization vari‐\n","                 ables that are unset or null. (See the Base Definitions  vol‐\n","                 ume  of POSIX.1‐2017, _\bS_\be_\bc_\bt_\bi_\bo_\bn _\b8_\b._\b2, _\bI_\bn_\bt_\be_\br_\bn_\ba_\bt_\bi_\bo_\bn_\ba_\bl_\bi_\bz_\ba_\bt_\bi_\bo_\bn _\bV_\ba_\br_\bi_\b‐\n","                 _\ba_\bb_\bl_\be_\bs for the precedence  of  internationalization  variables\n","                 used to determine the values of locale categories.)\n","\n","       _\bL_\bC_\b__\bA_\bL_\bL    If  set  to  a non-empty string value, override the values of\n","                 all the other internationalization variables.\n","\n","       _\bL_\bC_\b__\bC_\bT_\bY_\bP_\bE  Determine the locale for the interpretation of  sequences  of\n","                 bytes of text data as characters (for example, single-byte as\n","                 opposed to  multi-byte  characters  in  arguments  and  input\n","                 files).\n","\n","       _\bL_\bC_\b__\bM_\bE_\bS_\bS_\bA_\bG_\bE_\bS\n","                 Determine the locale that should be used to affect the format\n","                 and contents of diagnostic messages written to  standard  er‐\n","                 ror.\n","\n","       _\bN_\bL_\bS_\bP_\bA_\bT_\bH   Determine the location of message catalogs for the processing\n","                 of _\bL_\bC_\b__\bM_\bE_\bS_\bS_\bA_\bG_\bE_\bS.\n","\n","A\bAS\bSY\bYN\bNC\bCH\bHR\bRO\bON\bNO\bOU\bUS\bS E\bEV\bVE\bEN\bNT\bTS\bS\n","       Default.\n","\n","S\bST\bTD\bDO\bOU\bUT\bT\n","       The standard output shall contain  designated  portions  of  the  input\n","       files.\n","\n","       If  multiple _\bf_\bi_\bl_\be operands are specified, _\bh_\be_\ba_\bd shall precede the output\n","       for each with the header:\n","\n","           \"\\n==> %s <==\\n\", <_\bp_\ba_\bt_\bh_\bn_\ba_\bm_\be>\n","\n","       except that the first header written  shall  not  include  the  initial\n","       <newline>.\n","\n","S\bST\bTD\bDE\bER\bRR\bR\n","       The standard error shall be used only for diagnostic messages.\n","\n","O\bOU\bUT\bTP\bPU\bUT\bT F\bFI\bIL\bLE\bES\bS\n","       None.\n","\n","E\bEX\bXT\bTE\bEN\bND\bDE\bED\bD D\bDE\bES\bSC\bCR\bRI\bIP\bPT\bTI\bIO\bON\bN\n","       None.\n","\n","E\bEX\bXI\bIT\bT S\bST\bTA\bAT\bTU\bUS\bS\n","       The following exit values shall be returned:\n","\n","        0    Successful completion.\n","\n","       >0    An error occurred.\n","\n","C\bCO\bON\bNS\bSE\bEQ\bQU\bUE\bEN\bNC\bCE\bES\bS O\bOF\bF E\bER\bRR\bRO\bOR\bRS\bS\n","       Default.\n","\n","       _\bT_\bh_\be _\bf_\bo_\bl_\bl_\bo_\bw_\bi_\bn_\bg _\bs_\be_\bc_\bt_\bi_\bo_\bn_\bs _\ba_\br_\be _\bi_\bn_\bf_\bo_\br_\bm_\ba_\bt_\bi_\bv_\be_\b.\n","\n","A\bAP\bPP\bPL\bLI\bIC\bCA\bAT\bTI\bIO\bON\bN U\bUS\bSA\bAG\bGE\bE\n","       When using _\bh_\be_\ba_\bd to process pathnames, it is recommended that LC_ALL, or\n","       at least LC_CTYPE and LC_COLLATE, are set to POSIX or C in the environ‐\n","       ment, since pathnames can contain byte sequences that do not form valid\n","       characters in some locales, in which case the utility's behavior  would\n","       be  undefined.  In  the  POSIX  locale each byte is a valid single-byte\n","       character, and therefore this problem is avoided.\n","\n","E\bEX\bXA\bAM\bMP\bPL\bLE\bES\bS\n","       To write the first ten lines of all files (except those with a  leading\n","       period) in the directory:\n","\n","           head -- *\n","\n","R\bRA\bAT\bTI\bIO\bON\bNA\bAL\bLE\bE\n","       Although  it  is  possible  to  simulate _\bh_\be_\ba_\bd with _\bs_\be_\bd 10q for a single\n","       file, the standard developers decided that the popularity  of  _\bh_\be_\ba_\bd  on\n","       historical BSD systems warranted its inclusion alongside _\bt_\ba_\bi_\bl.\n","\n","       POSIX.1‐2008 version of _\bh_\be_\ba_\bd follows the Utility Syntax Guidelines. The\n","       -\b-n\bn option was added to this new interface so that _\bh_\be_\ba_\bd and  _\bt_\ba_\bi_\bl  would\n","       be  more logically related. Earlier versions of this standard allowed a\n","       -\b-n\bnu\bum\bmb\bbe\ber\br option. This form is no longer specified  by  POSIX.1‐2008  but\n","       may be present in some implementations.\n","\n","       There  is no -\b-c\bc option (as there is in _\bt_\ba_\bi_\bl) because it is not histori‐\n","       cal practice and because other utilities in this volume of POSIX.1‐2017\n","       provide similar functionality.\n","\n","F\bFU\bUT\bTU\bUR\bRE\bE D\bDI\bIR\bRE\bEC\bCT\bTI\bIO\bON\bNS\bS\n","       None.\n","\n","S\bSE\bEE\bE A\bAL\bLS\bSO\bO\n","       _\bs_\be_\bd, _\bt_\ba_\bi_\bl\n","\n","       The  Base  Definitions  volume  of POSIX.1‐2017, _\bC_\bh_\ba_\bp_\bt_\be_\br _\b8, _\bE_\bn_\bv_\bi_\br_\bo_\bn_\bm_\be_\bn_\bt\n","       _\bV_\ba_\br_\bi_\ba_\bb_\bl_\be_\bs, _\bS_\be_\bc_\bt_\bi_\bo_\bn _\b1_\b2_\b._\b2, _\bU_\bt_\bi_\bl_\bi_\bt_\by _\bS_\by_\bn_\bt_\ba_\bx _\bG_\bu_\bi_\bd_\be_\bl_\bi_\bn_\be_\bs\n","\n","C\bCO\bOP\bPY\bYR\bRI\bIG\bGH\bHT\bT\n","       Portions of this text are reprinted and reproduced in  electronic  form\n","       from  IEEE Std 1003.1-2017, Standard for Information Technology -- Por‐\n","       table Operating System Interface (POSIX), The Open Group Base  Specifi‐\n","       cations  Issue  7, 2018 Edition, Copyright (C) 2018 by the Institute of\n","       Electrical and Electronics Engineers, Inc and The Open Group.   In  the\n","       event of any discrepancy between this version and the original IEEE and\n","       The Open Group Standard, the original IEEE and The Open Group  Standard\n","       is  the  referee document. The original Standard can be obtained online\n","       at http://www.opengroup.org/unix/online.html .\n","\n","       Any typographical or formatting errors that appear  in  this  page  are\n","       most likely to have been introduced during the conversion of the source\n","       files to man page format. To report such errors,  see  https://www.ker‐\n","       nel.org/doc/man-pages/reporting_bugs.html .\n","\n","IEEE/The Open Group                  2017                         HEAD(1POSIX)\n"]}]},{"cell_type":"code","source":["#####\n","! cp /content/genome/human/Homo_sapiens.GRCh38.dna.chromosome.1.fa /content/genome/human/test.fa\n","! diff /content/genome/human/Homo_sapiens.GRCh38.dna.chromosome.1.fa /content/genome/human/test.fa"],"metadata":{"id":"O0BVETLvW9yl","executionInfo":{"status":"ok","timestamp":1751916583561,"user_tz":240,"elapsed":914,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}}},"execution_count":38,"outputs":[]},{"cell_type":"code","source":["! head -10 /content/genome/human/test.fa > /content/genome/human/test1.fa\n","! tail -10 /content/genome/human/test.fa > /content/genome/human/test2.fa"],"metadata":{"id":"OXeyCtBWXamh","executionInfo":{"status":"ok","timestamp":1751916638973,"user_tz":240,"elapsed":205,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}}},"execution_count":41,"outputs":[]},{"cell_type":"code","source":["! diff /content/genome/human/test1.fa /content/genome/human/test2.fa"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"hMml0TgFXdfa","executionInfo":{"status":"ok","timestamp":1751916644718,"user_tz":240,"elapsed":104,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"af286627-0108-47f7-ebf8-1042a094f109"},"execution_count":42,"outputs":[{"output_type":"stream","name":"stdout","text":["1d0\n","< >1 dna:chromosome chromosome:GRCh38:1:1:248956422:1 REF\n","10a10\n","> NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n"]}]},{"cell_type":"markdown","source":["##### **Attention**:  All the files downloaded and created using this notebook are temporary. If we turn off this notebook, they all disappeared.  What if we want to keep the downloaded files?"],"metadata":{"id":"4yzCo0n6ZKrI"}},{"cell_type":"code","source":["##### Connect to your google drive\n","from google.colab import drive\n","drive.mount('/content/drive')"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"woHgFuefyhPo","executionInfo":{"status":"ok","timestamp":1751917143604,"user_tz":240,"elapsed":19931,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"3546fc25-0631-4643-c970-a2ddc7df7e3c"},"execution_count":43,"outputs":[{"output_type":"stream","name":"stdout","text":["Mounted at /content/drive\n"]}]},{"cell_type":"code","source":["##### List the content of your google drive\n","! ls -al /content/drive/MyDrive"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"pnkcWopbZlvI","executionInfo":{"status":"ok","timestamp":1751917166525,"user_tz":240,"elapsed":503,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"cd589f5d-9a59-44e6-8459-9ddaf86227a3"},"execution_count":44,"outputs":[{"output_type":"stream","name":"stdout","text":["total 28\n","drwx------ 2 root root 4096 Sep 14  2024  Colab_BIO466\n","drwx------ 2 root root 4096 Apr  3 23:32  Colab_BIO485\n","drwx------ 2 root root 4096 Mar  2 16:50  Colab_Lib\n","drwx------ 2 root root 4096 Jan 14  2024 'Colab Notebooks'\n","drwx------ 2 root root 4096 Jan 22  2024  Collaborator\n","drwx------ 2 root root 4096 Sep 14  2024  Lab_share\n","drwx------ 2 root root 4096 Jan 22  2024  Miami\n"]}]},{"cell_type":"code","source":["##### Move the genome folder into your google drive\n","! mv /content/genome /content/drive/MyDrive/Colab_BIO466/"],"metadata":{"id":"1mae7YKKZ1WD","executionInfo":{"status":"ok","timestamp":1751917282664,"user_tz":240,"elapsed":7741,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}}},"execution_count":45,"outputs":[]},{"cell_type":"code","source":["! ls -al /content/drive/MyDrive/Colab_BIO466/"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"_zHhJJA7aGgO","executionInfo":{"status":"ok","timestamp":1751917320253,"user_tz":240,"elapsed":108,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"4657ec43-5670-4fce-f600-90102a4563b3"},"execution_count":47,"outputs":[{"output_type":"stream","name":"stdout","text":["total 16\n","drwx------  2 root root 4096 Sep 14  2024 colab_env\n","drwx------  2 root root 4096 Sep 14  2024 data\n","drwx------+ 3 root root 4096 Jul  7 19:41 genome\n","drwx------  2 root root 4096 Sep 15  2024 other_tools\n"]}]},{"cell_type":"code","source":["! ls -al /content/drive/MyDrive/Colab_BIO466/genome/human"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"FzClbG1LapAc","executionInfo":{"status":"ok","timestamp":1751918280757,"user_tz":240,"elapsed":104,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"c4919330-1399-476f-cbcd-792fcb0a5714"},"execution_count":59,"outputs":[{"output_type":"stream","name":"stdout","text":["total 556577\n","-rwx------+ 1 root root  63720222 Feb 15 15:49 Homo_sapiens.GRCh38.114.chromosome.1.gff3\n","-rwx------+ 1 root root 253105752 Jan 30 20:44 Homo_sapiens.GRCh38.dna.chromosome.1.fa\n","-rwx------+ 1 root root       605 Jul  7 19:30 test1.fa\n","-rwx------+ 1 root root       592 Jul  7 19:30 test2.fa\n","-rwx------+ 1 root root 253105752 Jul  7 19:29 test.fa\n"]}]}]}