{"cells":[{"cell_type":"code","execution_count":2,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":20521,"status":"ok","timestamp":1752622848917,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"},"user_tz":240},"id":"aQKqHagxKHgw","outputId":"c7993c4b-f780-487f-cb39-65304062dfc6"},"outputs":[{"output_type":"stream","name":"stdout","text":["Mounted at /content/drive\n"]}],"source":["# @title ### **Step 1: Mount Google Drive & Install Tools**\n","from google.colab import drive\n","drive.mount('/content/drive')"]},{"cell_type":"markdown","metadata":{"id":"f-WVvJzbKuts"},"source":["Follow the instruction https://github.com/FelixKrueger/TrimGalore\n","\n","Trim Galore is a a Perl wrapper around two tools: Cutadapt and FastQC. To use, ensure that these two pieces of software are available and copy the trim_galore script to a location available on the PATH"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"collapsed":true,"executionInfo":{"elapsed":10858,"status":"ok","timestamp":1752610896785,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"},"user_tz":240},"id":"L0n64G7iMJgC","outputId":"b74e6d42-0264-4e10-ac2b-5d01dc446281"},"outputs":[{"name":"stdout","output_type":"stream","text":["Requirement already satisfied: cutadapt in /usr/local/lib/python3.11/dist-packages (5.1)\n","Requirement already satisfied: dnaio>=1.2.3 in /usr/local/lib/python3.11/dist-packages (from cutadapt) (1.2.3)\n","Requirement already satisfied: xopen>=1.6.0 in /usr/local/lib/python3.11/dist-packages (from cutadapt) (2.0.2)\n","Requirement already satisfied: isal>=1.6.1 in /usr/local/lib/python3.11/dist-packages (from xopen>=1.6.0->cutadapt) (1.7.2)\n","Requirement already satisfied: zlib-ng>=0.4.1 in /usr/local/lib/python3.11/dist-packages (from xopen>=1.6.0->cutadapt) (0.5.1)\n","Reading package lists... Done\n","Building dependency tree... Done\n","Reading state information... Done\n","perl is already the newest version (5.34.0-3ubuntu1.4).\n","0 upgraded, 0 newly installed, 0 to remove and 31 not upgraded.\n","--2025-07-15 20:21:34--  https://github.com/FelixKrueger/TrimGalore/archive/refs/tags/0.6.10.tar.gz\n","Resolving github.com (github.com)... 140.82.116.4\n","Connecting to github.com (github.com)|140.82.116.4|:443... connected.\n","HTTP request sent, awaiting response... 302 Found\n","Location: https://codeload.github.com/FelixKrueger/TrimGalore/tar.gz/refs/tags/0.6.10 [following]\n","--2025-07-15 20:21:35--  https://codeload.github.com/FelixKrueger/TrimGalore/tar.gz/refs/tags/0.6.10\n","Resolving codeload.github.com (codeload.github.com)... 140.82.116.10\n","Connecting to codeload.github.com (codeload.github.com)|140.82.116.10|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: unspecified [application/x-gzip]\n","Saving to: ‘0.6.10.tar.gz’\n","\n","0.6.10.tar.gz           [     <=>            ]  25.59M  28.2MB/s    in 0.9s    \n","\n","2025-07-15 20:21:36 (28.2 MB/s) - ‘0.6.10.tar.gz’ saved [26831358]\n","\n","/content/TrimGalore-0.6.10/trim_galore\n","\n","                        Quality-/Adapter-/RRBS-/Speciality-Trimming\n","                                [powered by Cutadapt]\n","                                  version 0.6.10\n","\n","                               Last update: 02 02 2023\n","\n"]}],"source":["# 1. Install dependencies\n","!pip install cutadapt\n","!apt-get install perl\n","\n","# 2. Download and install TrimGalore\n","!wget https://github.com/FelixKrueger/TrimGalore/archive/refs/tags/0.6.10.tar.gz\n","!tar -xzf 0.6.10.tar.gz\n","\n","# 3. Make the script executable and add to PATH\n","import os\n","os.environ['PATH'] += os.pathsep + '/content/TrimGalore-0.6.10'\n","\n","# 4. Verify installation\n","!ls /content/TrimGalore-0.6.10/trim_galore  # Check if file exists\n","!/content/TrimGalore-0.6.10/trim_galore --version"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"collapsed":true,"executionInfo":{"elapsed":1916,"status":"ok","timestamp":1752610921785,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"},"user_tz":240},"id":"DSOMQLrtND0O","outputId":"b4b5cd7b-6dd2-4a45-f3f1-dbcfa868dee6"},"outputs":[{"name":"stdout","output_type":"stream","text":["total 4016603\n","-rw------- 1 root root 348532629 Jul  9 01:13 YapPool_S01_sub_1.fq.gz\n","-rw------- 1 root root 352131287 Jul  9 01:16 YapPool_S01_sub_2.fq.gz\n","-rw------- 1 root root 355127133 Jul  9 01:29 YapPool_S02_sub_1.fq.gz\n","-rw------- 1 root root 359222563 Jul  9 01:32 YapPool_S02_sub_2.fq.gz\n","-rw------- 1 root root 304346742 Jul  9 01:45 YapPool_S03_sub_1.fq.gz\n","-rw------- 1 root root 308548182 Jul  9 01:47 YapPool_S03_sub_2.fq.gz\n","-rw------- 1 root root 364095003 Jul  9 01:59 YapPool_S09_sub_1.fq.gz\n","-rw------- 1 root root 367720065 Jul  9 02:02 YapPool_S09_sub_2.fq.gz\n","-rw------- 1 root root 345373866 Jul  9 02:16 YapPool_S10_sub_1.fq.gz\n","-rw------- 1 root root 349413817 Jul  9 02:19 YapPool_S10_sub_2.fq.gz\n","-rw------- 1 root root 327577645 Jul  9 02:31 YapPool_S11_sub_1.fq.gz\n","-rw------- 1 root root 330909050 Jul  9 02:35 YapPool_S11_sub_2.fq.gz\n"]}],"source":["! ls -al /content/drive/MyDrive/Lab_share/Lab_data/class_466/raw.small.RNA-Seq/"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":1047297,"status":"ok","timestamp":1752612288646,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"},"user_tz":240},"id":"doi6VMzXOg75","outputId":"c9806e05-d8ce-457f-fccf-a6ee8606c491"},"outputs":[{"name":"stdout","output_type":"stream","text":["Multicore support not enabled. Proceeding with single-core trimming.\n","Path to Cutadapt set as: 'cutadapt' (default)\n","Cutadapt seems to be working fine (tested command 'cutadapt --version')\n","Cutadapt version: 5.1\n","single-core operation.\n","Proceeding with 'gzip' for decompression\n","To decrease CPU usage of decompression, please install 'igzip' and run again\n","\n","No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n","\n","Output will be written into the directory: /content/\n","Writing report to '/content/YapPool_S01_sub_1.fq.gz_trimming_report.txt'\n","\n","SUMMARISING RUN PARAMETERS\n","==========================\n","Input filename: drive/MyDrive/Lab_share/Lab_data/class_466/raw.small.RNA-Seq/YapPool_S01_sub_1.fq.gz\n","Trimming mode: paired-end\n","Trim Galore version: 0.6.10\n","Cutadapt version: 5.1\n","Number of cores used for trimming: 1\n","Quality Phred score cutoff: 20\n","Quality encoding type selected: ASCII+33\n","Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; user defined)\n","Maximum trimming error rate: 0.1 (default)\n","Minimum required adapter overlap (stringency): 1 bp\n","Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n","All Read 1 sequences will be trimmed by 2 bp from their 5' end to avoid poor qualities or biases\n","All Read 2 sequences will be trimmed by 2 bp from their 5' end to avoid poor qualities or biases (e.g. M-bias for BS-Seq applications)\n","All Read 1 sequences will be trimmed by 2 bp from their 3' end to avoid poor qualities or biases\n","All Read 2 sequences will be trimmed by 2 bp from their 3' end to avoid poor qualities or biases\n","Running FastQC on the data once trimming has completed\n","Output file(s) will be GZIP compressed\n","\n","Cutadapt seems to be fairly up-to-date (version 5.1). Setting -j 1\n","Writing final adapter and quality trimmed output to YapPool_S01_sub_1_trimmed.fq.gz\n","\n","\n","  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file drive/MyDrive/Lab_share/Lab_data/class_466/raw.small.RNA-Seq/YapPool_S01_sub_1.fq.gz <<< \n","This is cutadapt 5.1 with Python 3.11.13\n","Command line parameters: -j 1 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC drive/MyDrive/Lab_share/Lab_data/class_466/raw.small.RNA-Seq/YapPool_S01_sub_1.fq.gz\n","Processing single-end reads on 1 core ...\n","\n","=== Summary ===\n","\n","Total reads processed:               5,605,279\n","Reads with adapters:                 2,679,013 (47.8%)\n","Reads written (passing filters):     5,605,279 (100.0%)\n","\n","Total basepairs processed:   840,791,850 bp\n","Quality-trimmed:               7,361,624 bp (0.9%)\n","Total written (filtered):    794,723,499 bp (94.5%)\n","\n","=== Adapter 1 ===\n","\n","Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 2679013 times\n","\n","Minimum overlap: 1\n","No. of allowed errors:\n","1-9 bp: 0; 10-13 bp: 1\n","\n","Bases preceding removed adapters:\n","  A: 22.1%\n","  C: 32.1%\n","  G: 27.6%\n","  T: 16.4%\n","  none/other: 1.8%\n","\n","Overview of removed sequences\n","length\tcount\texpect\tmax.err\terror counts\n","1\t999472\t1401319.8\t0\t999472\n","2\t332683\t350329.9\t0\t332683\n","3\t111070\t87582.5\t0\t111070\n","4\t49003\t21895.6\t0\t49003\n","5\t36814\t5473.9\t0\t36814\n","6\t32356\t1368.5\t0\t32356\n","7\t32074\t342.1\t0\t32074\n","8\t31044\t85.5\t0\t31044\n","9\t31651\t21.4\t0\t31512 139\n","10\t31034\t5.3\t1\t30648 386\n","11\t31528\t1.3\t1\t31095 433\n","12\t30166\t0.3\t1\t29754 412\n","13\t30504\t0.1\t1\t30037 467\n","14\t29963\t0.1\t1\t29576 387\n","15\t29905\t0.1\t1\t29431 474\n","16\t29237\t0.1\t1\t28796 441\n","17\t29686\t0.1\t1\t29200 486\n","18\t27821\t0.1\t1\t27429 392\n","19\t28101\t0.1\t1\t27716 385\n","20\t27086\t0.1\t1\t26683 403\n","21\t27464\t0.1\t1\t26985 479\n","22\t26463\t0.1\t1\t26099 364\n","23\t25823\t0.1\t1\t25450 373\n","24\t25209\t0.1\t1\t24857 352\n","25\t24819\t0.1\t1\t24438 381\n","26\t24423\t0.1\t1\t24087 336\n","27\t23821\t0.1\t1\t23467 354\n","28\t23401\t0.1\t1\t23050 351\n","29\t22643\t0.1\t1\t22309 334\n","30\t21536\t0.1\t1\t21233 303\n","31\t21363\t0.1\t1\t21047 316\n","32\t21274\t0.1\t1\t20933 341\n","33\t19752\t0.1\t1\t19451 301\n","34\t19521\t0.1\t1\t19221 300\n","35\t18590\t0.1\t1\t18303 287\n","36\t17837\t0.1\t1\t17573 264\n","37\t17644\t0.1\t1\t17312 332\n","38\t17203\t0.1\t1\t16920 283\n","39\t15575\t0.1\t1\t15306 269\n","40\t15431\t0.1\t1\t15106 325\n","41\t16013\t0.1\t1\t15497 516\n","42\t17051\t0.1\t1\t16602 449\n","43\t13487\t0.1\t1\t13202 285\n","44\t13083\t0.1\t1\t12865 218\n","45\t10716\t0.1\t1\t10437 279\n","46\t12055\t0.1\t1\t11750 305\n","47\t12852\t0.1\t1\t12479 373\n","48\t14403\t0.1\t1\t14054 349\n","49\t14919\t0.1\t1\t14456 463\n","50\t28659\t0.1\t1\t28232 427\n","51\t25918\t0.1\t1\t25637 281\n","52\t12709\t0.1\t1\t12485 224\n","53\t12399\t0.1\t1\t12258 141\n","54\t8774\t0.1\t1\t8695 79\n","55\t2375\t0.1\t1\t2335 40\n","56\t882\t0.1\t1\t861 21\n","57\t474\t0.1\t1\t467 7\n","58\t227\t0.1\t1\t220 7\n","59\t164\t0.1\t1\t160 4\n","60\t65\t0.1\t1\t58 7\n","61\t53\t0.1\t1\t42 11\n","62\t28\t0.1\t1\t22 6\n","63\t14\t0.1\t1\t7 7\n","64\t11\t0.1\t1\t6 5\n","65\t11\t0.1\t1\t7 4\n","66\t17\t0.1\t1\t7 10\n","67\t17\t0.1\t1\t11 6\n","68\t28\t0.1\t1\t16 12\n","69\t35\t0.1\t1\t7 28\n","70\t15\t0.1\t1\t5 10\n","71\t15\t0.1\t1\t4 11\n","72\t15\t0.1\t1\t8 7\n","73\t11\t0.1\t1\t7 4\n","74\t19\t0.1\t1\t9 10\n","75\t20\t0.1\t1\t12 8\n","76\t14\t0.1\t1\t7 7\n","77\t30\t0.1\t1\t19 11\n","78\t26\t0.1\t1\t20 6\n","79\t26\t0.1\t1\t25 1\n","80\t23\t0.1\t1\t17 6\n","81\t35\t0.1\t1\t25 10\n","82\t29\t0.1\t1\t25 4\n","83\t36\t0.1\t1\t29 7\n","84\t28\t0.1\t1\t24 4\n","85\t31\t0.1\t1\t27 4\n","86\t30\t0.1\t1\t25 5\n","87\t26\t0.1\t1\t21 5\n","88\t34\t0.1\t1\t24 10\n","89\t35\t0.1\t1\t21 14\n","90\t37\t0.1\t1\t35 2\n","91\t44\t0.1\t1\t36 8\n","92\t28\t0.1\t1\t26 2\n","93\t38\t0.1\t1\t33 5\n","94\t29\t0.1\t1\t28 1\n","95\t49\t0.1\t1\t42 7\n","96\t33\t0.1\t1\t31 2\n","97\t61\t0.1\t1\t52 9\n","98\t69\t0.1\t1\t51 18\n","99\t90\t0.1\t1\t75 15\n","100\t53\t0.1\t1\t47 6\n","101\t59\t0.1\t1\t53 6\n","102\t61\t0.1\t1\t53 8\n","103\t53\t0.1\t1\t46 7\n","104\t39\t0.1\t1\t37 2\n","105\t48\t0.1\t1\t42 6\n","106\t55\t0.1\t1\t46 9\n","107\t28\t0.1\t1\t26 2\n","108\t41\t0.1\t1\t36 5\n","109\t46\t0.1\t1\t41 5\n","110\t32\t0.1\t1\t27 5\n","111\t36\t0.1\t1\t32 4\n","112\t29\t0.1\t1\t25 4\n","113\t31\t0.1\t1\t28 3\n","114\t37\t0.1\t1\t31 6\n","115\t39\t0.1\t1\t33 6\n","116\t36\t0.1\t1\t28 8\n","117\t39\t0.1\t1\t35 4\n","118\t51\t0.1\t1\t44 7\n","119\t33\t0.1\t1\t25 8\n","120\t39\t0.1\t1\t23 16\n","121\t43\t0.1\t1\t30 13\n","122\t38\t0.1\t1\t30 8\n","123\t43\t0.1\t1\t29 14\n","124\t28\t0.1\t1\t19 9\n","125\t27\t0.1\t1\t20 7\n","126\t55\t0.1\t1\t39 16\n","127\t43\t0.1\t1\t34 9\n","128\t43\t0.1\t1\t33 10\n","129\t60\t0.1\t1\t40 20\n","130\t51\t0.1\t1\t42 9\n","131\t63\t0.1\t1\t55 8\n","132\t58\t0.1\t1\t44 14\n","133\t67\t0.1\t1\t58 9\n","134\t75\t0.1\t1\t63 12\n","135\t70\t0.1\t1\t56 14\n","136\t101\t0.1\t1\t85 16\n","137\t87\t0.1\t1\t79 8\n","138\t96\t0.1\t1\t81 15\n","139\t68\t0.1\t1\t64 4\n","140\t105\t0.1\t1\t99 6\n","141\t188\t0.1\t1\t176 12\n","142\t88\t0.1\t1\t80 8\n","143\t105\t0.1\t1\t100 5\n","144\t164\t0.1\t1\t148 16\n","145\t191\t0.1\t1\t168 23\n","146\t223\t0.1\t1\t202 21\n","147\t390\t0.1\t1\t363 27\n","148\t169\t0.1\t1\t107 62\n","149\t543\t0.1\t1\t5 538\n","150\t45441\t0.1\t1\t82 45359\n","\n","RUN STATISTICS FOR INPUT FILE: drive/MyDrive/Lab_share/Lab_data/class_466/raw.small.RNA-Seq/YapPool_S01_sub_1.fq.gz\n","=============================================\n","5605279 sequences processed in total\n","The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n","\n","Writing report to '/content/YapPool_S01_sub_2.fq.gz_trimming_report.txt'\n","\n","SUMMARISING RUN PARAMETERS\n","==========================\n","Input filename: drive/MyDrive/Lab_share/Lab_data/class_466/raw.small.RNA-Seq/YapPool_S01_sub_2.fq.gz\n","Trimming mode: paired-end\n","Trim Galore version: 0.6.10\n","Cutadapt version: 5.1\n","Number of cores used for trimming: 1\n","Quality Phred score cutoff: 20\n","Quality encoding type selected: ASCII+33\n","Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; user defined)\n","Maximum trimming error rate: 0.1 (default)\n","Minimum required adapter overlap (stringency): 1 bp\n","Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n","All Read 1 sequences will be trimmed by 2 bp from their 5' end to avoid poor qualities or biases\n","All Read 2 sequences will be trimmed by 2 bp from their 5' end to avoid poor qualities or biases (e.g. M-bias for BS-Seq applications)\n","All Read 1 sequences will be trimmed by 2 bp from their 3' end to avoid poor qualities or biases\n","All Read 2 sequences will be trimmed by 2 bp from their 3' end to avoid poor qualities or biases\n","Running FastQC on the data once trimming has completed\n","Output file(s) will be GZIP compressed\n","\n","Cutadapt seems to be fairly up-to-date (version 5.1). Setting -j -j 1\n","Writing final adapter and quality trimmed output to YapPool_S01_sub_2_trimmed.fq.gz\n","\n","\n","  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file drive/MyDrive/Lab_share/Lab_data/class_466/raw.small.RNA-Seq/YapPool_S01_sub_2.fq.gz <<< \n","This is cutadapt 5.1 with Python 3.11.13\n","Command line parameters: -j 1 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC drive/MyDrive/Lab_share/Lab_data/class_466/raw.small.RNA-Seq/YapPool_S01_sub_2.fq.gz\n","Processing single-end reads on 1 core ...\n","\n","=== Summary ===\n","\n","Total reads processed:               5,605,279\n","Reads with adapters:                 2,723,985 (48.6%)\n","Reads written (passing filters):     5,605,279 (100.0%)\n","\n","Total basepairs processed:   840,791,850 bp\n","Quality-trimmed:              10,783,043 bp (1.3%)\n","Total written (filtered):    798,166,168 bp (94.9%)\n","\n","=== Adapter 1 ===\n","\n","Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 2723985 times\n","\n","Minimum overlap: 1\n","No. of allowed errors:\n","1-9 bp: 0; 10-13 bp: 1\n","\n","Bases preceding removed adapters:\n","  A: 24.7%\n","  C: 30.1%\n","  G: 32.3%\n","  T: 13.0%\n","  none/other: 0.0%\n","\n","Overview of removed sequences\n","length\tcount\texpect\tmax.err\terror counts\n","1\t1056283\t1401319.8\t0\t1056283\n","2\t346103\t350329.9\t0\t346103\n","3\t132272\t87582.5\t0\t132272\n","4\t53052\t21895.6\t0\t53052\n","5\t37439\t5473.9\t0\t37439\n","6\t32941\t1368.5\t0\t32941\n","7\t32220\t342.1\t0\t32220\n","8\t31357\t85.5\t0\t31357\n","9\t31571\t21.4\t0\t31441 130\n","10\t31466\t5.3\t1\t31045 421\n","11\t31762\t1.3\t1\t31299 463\n","12\t31370\t0.3\t1\t30931 439\n","13\t29866\t0.1\t1\t29536 330\n","14\t30529\t0.1\t1\t30130 399\n","15\t30209\t0.1\t1\t29795 414\n","16\t29186\t0.1\t1\t28815 371\n","17\t29900\t0.1\t1\t29468 432\n","18\t29188\t0.1\t1\t28714 474\n","19\t34772\t0.1\t1\t34038 734\n","20\t28479\t0.1\t1\t28111 368\n","21\t24874\t0.1\t1\t24656 218\n","22\t24277\t0.1\t1\t24078 199\n","23\t25537\t0.1\t1\t25283 254\n","24\t30468\t0.1\t1\t30132 336\n","25\t21064\t0.1\t1\t20870 194\n","26\t25188\t0.1\t1\t24967 221\n","27\t32895\t0.1\t1\t32597 298\n","28\t18756\t0.1\t1\t18519 237\n","29\t27924\t0.1\t1\t27691 233\n","30\t19247\t0.1\t1\t19034 213\n","31\t21867\t0.1\t1\t21679 188\n","32\t17552\t0.1\t1\t17398 154\n","33\t34142\t0.1\t1\t33941 201\n","34\t12576\t0.1\t1\t12436 140\n","35\t14585\t0.1\t1\t14438 147\n","36\t24861\t0.1\t1\t24732 129\n","37\t13322\t0.1\t1\t13205 117\n","38\t17840\t0.1\t1\t17734 106\n","39\t12517\t0.1\t1\t12417 100\n","40\t25779\t0.1\t1\t25679 100\n","41\t5472\t0.1\t1\t5389 83\n","42\t13006\t0.1\t1\t12931 75\n","43\t16348\t0.1\t1\t16265 83\n","44\t8163\t0.1\t1\t8091 72\n","45\t14613\t0.1\t1\t14530 83\n","46\t19204\t0.1\t1\t19117 87\n","47\t9029\t0.1\t1\t8948 81\n","48\t9365\t0.1\t1\t9320 45\n","49\t6292\t0.1\t1\t6233 59\n","50\t11794\t0.1\t1\t11751 43\n","51\t3003\t0.1\t1\t2959 44\n","52\t9419\t0.1\t1\t9375 44\n","53\t4913\t0.1\t1\t4875 38\n","54\t4662\t0.1\t1\t4626 36\n","55\t7290\t0.1\t1\t7251 39\n","56\t3230\t0.1\t1\t3196 34\n","57\t2047\t0.1\t1\t2024 23\n","58\t5080\t0.1\t1\t5052 28\n","59\t819\t0.1\t1\t803 16\n","60\t2304\t0.1\t1\t2287 17\n","61\t2518\t0.1\t1\t2497 21\n","62\t3160\t0.1\t1\t3137 23\n","63\t3568\t0.1\t1\t3551 17\n","64\t342\t0.1\t1\t331 11\n","65\t895\t0.1\t1\t873 22\n","66\t878\t0.1\t1\t860 18\n","67\t1273\t0.1\t1\t1251 22\n","68\t1152\t0.1\t1\t1131 21\n","69\t1071\t0.1\t1\t1053 18\n","70\t1024\t0.1\t1\t1020 4\n","71\t866\t0.1\t1\t855 11\n","72\t989\t0.1\t1\t980 9\n","73\t895\t0.1\t1\t886 9\n","74\t797\t0.1\t1\t785 12\n","75\t811\t0.1\t1\t800 11\n","76\t744\t0.1\t1\t729 15\n","77\t762\t0.1\t1\t752 10\n","78\t669\t0.1\t1\t662 7\n","79\t624\t0.1\t1\t617 7\n","80\t527\t0.1\t1\t522 5\n","81\t524\t0.1\t1\t515 9\n","82\t454\t0.1\t1\t442 12\n","83\t429\t0.1\t1\t410 19\n","84\t392\t0.1\t1\t379 13\n","85\t368\t0.1\t1\t357 11\n","86\t372\t0.1\t1\t359 13\n","87\t319\t0.1\t1\t305 14\n","88\t273\t0.1\t1\t263 10\n","89\t288\t0.1\t1\t283 5\n","90\t269\t0.1\t1\t260 9\n","91\t245\t0.1\t1\t229 16\n","92\t183\t0.1\t1\t181 2\n","93\t187\t0.1\t1\t182 5\n","94\t143\t0.1\t1\t137 6\n","95\t139\t0.1\t1\t135 4\n","96\t104\t0.1\t1\t100 4\n","97\t144\t0.1\t1\t137 7\n","98\t109\t0.1\t1\t104 5\n","99\t123\t0.1\t1\t116 7\n","100\t99\t0.1\t1\t94 5\n","101\t91\t0.1\t1\t85 6\n","102\t91\t0.1\t1\t79 12\n","103\t68\t0.1\t1\t62 6\n","104\t54\t0.1\t1\t48 6\n","105\t61\t0.1\t1\t55 6\n","106\t66\t0.1\t1\t54 12\n","107\t45\t0.1\t1\t34 11\n","108\t56\t0.1\t1\t45 11\n","109\t72\t0.1\t1\t50 22\n","110\t40\t0.1\t1\t38 2\n","111\t40\t0.1\t1\t38 2\n","112\t43\t0.1\t1\t34 9\n","113\t40\t0.1\t1\t33 7\n","114\t42\t0.1\t1\t33 9\n","115\t45\t0.1\t1\t36 9\n","116\t34\t0.1\t1\t30 4\n","117\t41\t0.1\t1\t35 6\n","118\t52\t0.1\t1\t44 8\n","119\t40\t0.1\t1\t30 10\n","120\t35\t0.1\t1\t26 9\n","121\t56\t0.1\t1\t30 26\n","122\t37\t0.1\t1\t31 6\n","123\t39\t0.1\t1\t31 8\n","124\t33\t0.1\t1\t26 7\n","125\t47\t0.1\t1\t28 19\n","126\t46\t0.1\t1\t43 3\n","127\t42\t0.1\t1\t36 6\n","128\t43\t0.1\t1\t32 11\n","129\t74\t0.1\t1\t46 28\n","130\t54\t0.1\t1\t43 11\n","131\t66\t0.1\t1\t57 9\n","132\t55\t0.1\t1\t45 10\n","133\t68\t0.1\t1\t60 8\n","134\t68\t0.1\t1\t65 3\n","135\t77\t0.1\t1\t57 20\n","136\t101\t0.1\t1\t89 12\n","137\t96\t0.1\t1\t84 12\n","138\t99\t0.1\t1\t94 5\n","139\t82\t0.1\t1\t70 12\n","140\t112\t0.1\t1\t103 9\n","141\t198\t0.1\t1\t185 13\n","142\t86\t0.1\t1\t84 2\n","143\t110\t0.1\t1\t104 6\n","144\t161\t0.1\t1\t154 7\n","145\t188\t0.1\t1\t181 7\n","146\t214\t0.1\t1\t208 6\n","147\t392\t0.1\t1\t369 23\n","148\t120\t0.1\t1\t110 10\n","149\t11\t0.1\t1\t5 6\n","150\t136\t0.1\t1\t85 51\n","\n","RUN STATISTICS FOR INPUT FILE: drive/MyDrive/Lab_share/Lab_data/class_466/raw.small.RNA-Seq/YapPool_S01_sub_2.fq.gz\n","=============================================\n","5605279 sequences processed in total\n","The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n","\n","Validate paired-end files YapPool_S01_sub_1_trimmed.fq.gz and YapPool_S01_sub_2_trimmed.fq.gz\n","file_1: YapPool_S01_sub_1_trimmed.fq.gz, file_2: YapPool_S01_sub_2_trimmed.fq.gz\n","\n","\n",">>>>> Now validing the length of the 2 paired-end infiles: YapPool_S01_sub_1_trimmed.fq.gz and YapPool_S01_sub_2_trimmed.fq.gz <<<<<\n","Writing validated paired-end Read 1 reads to YapPool_S01_sub_1_val_1.fq.gz\n","Writing validated paired-end Read 2 reads to YapPool_S01_sub_2_val_2.fq.gz\n","\n","Total number of sequences analysed: 5605279\n","\n","Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 75357 (1.34%)\n","\n","\n","  >>> Now running FastQC on the validated data YapPool_S01_sub_1_val_1.fq.gz<<<\n","\n","Can't exec \"fastqc\": No such file or directory at /content/TrimGalore-0.6.10/trim_galore line 1555, <IN2> line 22421116.\n","\n","  >>> Now running FastQC on the validated data YapPool_S01_sub_2_val_2.fq.gz<<<\n","\n","Can't exec \"fastqc\": No such file or directory at /content/TrimGalore-0.6.10/trim_galore line 1565, <IN2> line 22421116.\n","Deleting both intermediate output files YapPool_S01_sub_1_trimmed.fq.gz and YapPool_S01_sub_2_trimmed.fq.gz\n","\n","====================================================================================================\n","\n"]}],"source":["##### This process is done within 5 minutes\n","!trim_galore \\\n","--paired \\\n","--quality 20 \\\n","--length 20 \\\n","--gzip \\\n","--output_dir ./ \\\n","--fastqc \\\n","--clip_R1 2 \\\n","--clip_R2 2 \\\n","--three_prime_clip_R1 2 \\\n","--three_prime_clip_R2 2 \\\n","--illumina \\\n","drive/MyDrive/Lab_share/Lab_data/class_466/raw.small.RNA-Seq/YapPool_S01_sub_1.fq.gz \\\n","drive/MyDrive/Lab_share/Lab_data/class_466/raw.small.RNA-Seq/YapPool_S01_sub_2.fq.gz"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"collapsed":true,"executionInfo":{"elapsed":43,"status":"ok","timestamp":1752612332039,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"},"user_tz":240},"id":"AnYD8QgzPkMo","outputId":"df58b8ae-49bb-47d0-a270-078bfcbdb685"},"outputs":[{"name":"stdout","output_type":"stream","text":["Writing run_trimgalore.sh\n"]}],"source":["%%writefile run_trimgalore.sh\n","#!/bin/bash\n","\n","# Exit immediately if a command exits with a non-zero status.\n","set -e\n","\n","echo \"Starting TrimGalore process...\"\n","\n","# --- Configuration for inputs and outputs ---\n","SEQ_DIR='drive/MyDrive/Lab_share/Lab_data/class_466/raw.small.RNA-Seq'\n","OUT_DIR='drive/MyDrive/Lab_share/Lab_data/class_466/clean.trimgalore.RNA-Seq'\n","# Define samples as a bash array\n","SAMPLES=(\"YapPool_S01\" \"YapPool_S02\" \"YapPool_S03\" \"YapPool_S09\" \"YapPool_S10\" \"YapPool_S11\")\n","\n","# Create the output directory if it doesn't exist\n","mkdir -p ${OUT_DIR}\n","echo \"Ensured output directory exists: ${OUT_DIR}\"\n","\n","# Loop through each sample and run TrimGalore\n","for SAMPLE in \"${SAMPLES[@]}\"; do\n","    echo \"Processing sample: ${SAMPLE}\"\n","\n","    trim_galore \\\n","    --paired \\\n","    --quality 20 \\\n","    --length 20 \\\n","    --gzip \\\n","    --output_dir ${OUT_DIR} \\\n","    --clip_R1 2 \\\n","    --clip_R2 2 \\\n","    --three_prime_clip_R1 2 \\\n","    --three_prime_clip_R2 2 \\\n","    --illumina \\\n","    --basename ${SAMPLE}_trim_ \\\n","    ${SEQ_DIR}/${SAMPLE}_sub_1.fq.gz \\\n","    ${SEQ_DIR}/${SAMPLE}_sub_2.fq.gz \\\n","    > ${OUT_DIR}/${SAMPLE}_trim.log 2>&1\n","\n","    echo \"Finished processing ${SAMPLE}\"\n","done"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"sxfOUJZo2iSS"},"outputs":[],"source":["! chmod +x run_trimgalore.sh"]},{"cell_type":"code","execution_count":4,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"sK1QYVmn1fpS","executionInfo":{"status":"ok","timestamp":1752628390326,"user_tz":240,"elapsed":113,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"3aa1fd2d-f015-4df2-a87c-cac519f24b93"},"outputs":[{"output_type":"stream","name":"stdout","text":["/bin/bash: line 1: ./run_trimgalore.sh: No such file or directory\n","\n","real\t0m0.002s\n","user\t0m0.001s\n","sys\t0m0.001s\n"]}],"source":["# This took 140 minutes to finish\n","! time ./run_trimgalore.sh"]},{"cell_type":"code","execution_count":6,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":129,"status":"ok","timestamp":1752628439780,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"},"user_tz":240},"id":"t2KNYHGIPokB","outputId":"d2c65613-ddc8-494b-80ea-5af1c3d9a215"},"outputs":[{"output_type":"stream","name":"stdout","text":["total 3761363\n","-rw------- 1 root root      5244 Jul 15 20:55 YapPool_S01_sub_1.fq.gz_trimming_report.txt\n","-rw------- 1 root root      5540 Jul 15 21:07 YapPool_S01_sub_2.fq.gz_trimming_report.txt\n","-rw------- 1 root root     13637 Jul 15 21:08 YapPool_S01_trim.log\n","-rw------- 1 root root 327476943 Jul 15 21:00 YapPool_S01_trim__val_1.fq.gz\n","-rw------- 1 root root 328406041 Jul 15 21:00 YapPool_S01_trim__val_2.fq.gz\n","-rw------- 1 root root      5251 Jul 15 21:13 YapPool_S02_sub_1.fq.gz_trimming_report.txt\n","-rw------- 1 root root      5535 Jul 15 21:25 YapPool_S02_sub_2.fq.gz_trimming_report.txt\n","-rw------- 1 root root     13639 Jul 15 21:25 YapPool_S02_trim.log\n","-rw------- 1 root root 334036068 Jul 15 21:18 YapPool_S02_trim__val_1.fq.gz\n","-rw------- 1 root root 334476109 Jul 15 21:18 YapPool_S02_trim__val_2.fq.gz\n","-rw------- 1 root root      5135 Jul 15 21:30 YapPool_S03_sub_1.fq.gz_trimming_report.txt\n","-rw------- 1 root root      5516 Jul 15 21:41 YapPool_S03_sub_2.fq.gz_trimming_report.txt\n","-rw------- 1 root root     13504 Jul 15 21:41 YapPool_S03_trim.log\n","-rw------- 1 root root 286745659 Jul 15 21:34 YapPool_S03_trim__val_1.fq.gz\n","-rw------- 1 root root 287838243 Jul 15 21:34 YapPool_S03_trim__val_2.fq.gz\n","-rw------- 1 root root      5179 Jul 15 21:46 YapPool_S09_sub_1.fq.gz_trimming_report.txt\n","-rw------- 1 root root      5530 Jul 15 21:59 YapPool_S09_sub_2.fq.gz_trimming_report.txt\n","-rw------- 1 root root     13562 Jul 15 21:59 YapPool_S09_trim.log\n","-rw------- 1 root root 341248598 Jul 15 21:51 YapPool_S09_trim__val_1.fq.gz\n","-rw------- 1 root root 341412534 Jul 15 21:51 YapPool_S09_trim__val_2.fq.gz\n","-rw------- 1 root root      5134 Jul 15 22:04 YapPool_S10_sub_1.fq.gz_trimming_report.txt\n","-rw------- 1 root root      5517 Jul 15 22:16 YapPool_S10_sub_2.fq.gz_trimming_report.txt\n","-rw------- 1 root root     13504 Jul 15 22:16 YapPool_S10_trim.log\n","-rw------- 1 root root 325975987 Jul 15 22:09 YapPool_S10_trim__val_1.fq.gz\n","-rw------- 1 root root 326857371 Jul 15 22:09 YapPool_S10_trim__val_2.fq.gz\n","-rw------- 1 root root      5146 Jul 15 22:21 YapPool_S11_sub_1.fq.gz_trimming_report.txt\n","-rw------- 1 root root      5510 Jul 15 22:32 YapPool_S11_sub_2.fq.gz_trimming_report.txt\n","-rw------- 1 root root     13509 Jul 15 22:33 YapPool_S11_trim.log\n","-rw------- 1 root root 308137243 Jul 15 22:25 YapPool_S11_trim__val_1.fq.gz\n","-rw------- 1 root root 308870036 Jul 15 22:25 YapPool_S11_trim__val_2.fq.gz\n"]}],"source":["! ls -al drive/MyDrive/Lab_share/Lab_data/class_466/clean.trimgalore.RNA-Seq/"]},{"cell_type":"code","execution_count":7,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"collapsed":true,"executionInfo":{"elapsed":29234,"status":"ok","timestamp":1752629313460,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"},"user_tz":240},"id":"bb78DZQSRfy2","outputId":"bd3aa4c5-a817-4115-e47e-1b3eea25f0a5"},"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting multiqc\n","  Downloading multiqc-1.30-py3-none-any.whl.metadata (46 kB)\n","\u001b[?25l     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m0.0/46.3 kB\u001b[0m \u001b[31m?\u001b[0m eta \u001b[36m-:--:--\u001b[0m\r\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m46.3/46.3 kB\u001b[0m \u001b[31m2.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hCollecting boto3 (from multiqc)\n","  Downloading boto3-1.39.5-py3-none-any.whl.metadata (6.6 kB)\n","Requirement already satisfied: click in /usr/local/lib/python3.11/dist-packages (from multiqc) (8.2.1)\n","Requirement already satisfied: humanize in /usr/local/lib/python3.11/dist-packages (from multiqc) (4.12.3)\n","Requirement already satisfied: importlib_metadata in /usr/local/lib/python3.11/dist-packages (from multiqc) (8.7.0)\n","Requirement already satisfied: jinja2>=3.0.0 in /usr/local/lib/python3.11/dist-packages (from multiqc) (3.1.6)\n","Collecting kaleido==0.2.1 (from multiqc)\n","  Downloading kaleido-0.2.1-py2.py3-none-manylinux1_x86_64.whl.metadata (15 kB)\n","Requirement already satisfied: markdown in /usr/local/lib/python3.11/dist-packages (from multiqc) (3.8.2)\n","Requirement already satisfied: numpy in /usr/local/lib/python3.11/dist-packages (from multiqc) (2.0.2)\n","Requirement already satisfied: packaging in /usr/local/lib/python3.11/dist-packages (from multiqc) (24.2)\n","Requirement already satisfied: requests in /usr/local/lib/python3.11/dist-packages (from multiqc) (2.32.3)\n","Requirement already satisfied: Pillow>=10 in /usr/local/lib/python3.11/dist-packages (from multiqc) (11.2.1)\n","Requirement already satisfied: plotly>=5.18 in /usr/local/lib/python3.11/dist-packages (from multiqc) (5.24.1)\n","Requirement already satisfied: pyyaml>=4 in /usr/local/lib/python3.11/dist-packages (from multiqc) (6.0.2)\n","Requirement already satisfied: rich>=10 in /usr/local/lib/python3.11/dist-packages (from multiqc) (13.9.4)\n","Collecting rich-click (from multiqc)\n","  Downloading rich_click-1.8.9-py3-none-any.whl.metadata (7.9 kB)\n","Collecting coloredlogs (from multiqc)\n","  Downloading coloredlogs-15.0.1-py2.py3-none-any.whl.metadata (12 kB)\n","Collecting spectra>=0.0.10 (from multiqc)\n","  Downloading spectra-0.1.0-py3-none-any.whl.metadata (927 bytes)\n","Requirement already satisfied: pydantic>=2.7.0 in /usr/local/lib/python3.11/dist-packages (from multiqc) (2.11.7)\n","Requirement already satisfied: typeguard in /usr/local/lib/python3.11/dist-packages (from multiqc) (4.4.4)\n","Requirement already satisfied: tqdm in /usr/local/lib/python3.11/dist-packages (from multiqc) (4.67.1)\n","Collecting python-dotenv (from multiqc)\n","  Downloading python_dotenv-1.1.1-py3-none-any.whl.metadata (24 kB)\n","Requirement already satisfied: natsort in /usr/local/lib/python3.11/dist-packages (from multiqc) (8.4.0)\n","Requirement already satisfied: tiktoken in /usr/local/lib/python3.11/dist-packages (from multiqc) (0.9.0)\n","Requirement already satisfied: jsonschema in /usr/local/lib/python3.11/dist-packages (from multiqc) (4.24.0)\n","Collecting polars-lts-cpu (from multiqc)\n","  Downloading polars_lts_cpu-1.31.0-cp39-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (14 kB)\n","Requirement already satisfied: pyarrow in /usr/local/lib/python3.11/dist-packages (from multiqc) (18.1.0)\n","Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.11/dist-packages (from jinja2>=3.0.0->multiqc) (3.0.2)\n","Requirement already satisfied: tenacity>=6.2.0 in /usr/local/lib/python3.11/dist-packages (from plotly>=5.18->multiqc) (8.5.0)\n","Requirement already satisfied: annotated-types>=0.6.0 in /usr/local/lib/python3.11/dist-packages (from pydantic>=2.7.0->multiqc) (0.7.0)\n","Requirement already satisfied: pydantic-core==2.33.2 in /usr/local/lib/python3.11/dist-packages (from pydantic>=2.7.0->multiqc) (2.33.2)\n","Requirement already satisfied: typing-extensions>=4.12.2 in /usr/local/lib/python3.11/dist-packages (from pydantic>=2.7.0->multiqc) (4.14.1)\n","Requirement already satisfied: typing-inspection>=0.4.0 in /usr/local/lib/python3.11/dist-packages (from pydantic>=2.7.0->multiqc) (0.4.1)\n","Requirement already satisfied: markdown-it-py>=2.2.0 in /usr/local/lib/python3.11/dist-packages (from rich>=10->multiqc) (3.0.0)\n","Requirement already satisfied: pygments<3.0.0,>=2.13.0 in /usr/local/lib/python3.11/dist-packages (from rich>=10->multiqc) (2.19.2)\n","Collecting colormath2>=3.0.0 (from spectra>=0.0.10->multiqc)\n","  Downloading colormath2-3.0.3-py3-none-any.whl.metadata (2.9 kB)\n","Collecting botocore<1.40.0,>=1.39.5 (from boto3->multiqc)\n","  Downloading botocore-1.39.5-py3-none-any.whl.metadata (5.7 kB)\n","Collecting jmespath<2.0.0,>=0.7.1 (from boto3->multiqc)\n","  Downloading jmespath-1.0.1-py3-none-any.whl.metadata (7.6 kB)\n","Collecting s3transfer<0.14.0,>=0.13.0 (from boto3->multiqc)\n","  Downloading s3transfer-0.13.0-py3-none-any.whl.metadata (1.7 kB)\n","Collecting humanfriendly>=9.1 (from coloredlogs->multiqc)\n","  Downloading humanfriendly-10.0-py2.py3-none-any.whl.metadata (9.2 kB)\n","Requirement already satisfied: zipp>=3.20 in /usr/local/lib/python3.11/dist-packages (from importlib_metadata->multiqc) (3.23.0)\n","Requirement already satisfied: attrs>=22.2.0 in /usr/local/lib/python3.11/dist-packages (from jsonschema->multiqc) (25.3.0)\n","Requirement already satisfied: jsonschema-specifications>=2023.03.6 in /usr/local/lib/python3.11/dist-packages (from jsonschema->multiqc) (2025.4.1)\n","Requirement already satisfied: referencing>=0.28.4 in /usr/local/lib/python3.11/dist-packages (from jsonschema->multiqc) (0.36.2)\n","Requirement already satisfied: rpds-py>=0.7.1 in /usr/local/lib/python3.11/dist-packages (from jsonschema->multiqc) (0.26.0)\n","Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.11/dist-packages (from requests->multiqc) (3.4.2)\n","Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.11/dist-packages (from requests->multiqc) (3.10)\n","Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.11/dist-packages (from requests->multiqc) (2.4.0)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.11/dist-packages (from requests->multiqc) (2025.7.9)\n","Requirement already satisfied: regex>=2022.1.18 in /usr/local/lib/python3.11/dist-packages (from tiktoken->multiqc) (2024.11.6)\n","Requirement already satisfied: python-dateutil<3.0.0,>=2.1 in /usr/local/lib/python3.11/dist-packages (from botocore<1.40.0,>=1.39.5->boto3->multiqc) (2.9.0.post0)\n","Requirement already satisfied: networkx>=2.0 in /usr/local/lib/python3.11/dist-packages (from colormath2>=3.0.0->spectra>=0.0.10->multiqc) (3.5)\n","Requirement already satisfied: mdurl~=0.1 in /usr/local/lib/python3.11/dist-packages (from markdown-it-py>=2.2.0->rich>=10->multiqc) (0.1.2)\n","Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.11/dist-packages (from python-dateutil<3.0.0,>=2.1->botocore<1.40.0,>=1.39.5->boto3->multiqc) (1.17.0)\n","Downloading multiqc-1.30-py3-none-any.whl (4.8 MB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m4.8/4.8 MB\u001b[0m \u001b[31m43.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading kaleido-0.2.1-py2.py3-none-manylinux1_x86_64.whl (79.9 MB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m79.9/79.9 MB\u001b[0m \u001b[31m8.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading spectra-0.1.0-py3-none-any.whl (17 kB)\n","Downloading boto3-1.39.5-py3-none-any.whl (139 kB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m139.9/139.9 kB\u001b[0m \u001b[31m8.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m46.0/46.0 kB\u001b[0m \u001b[31m2.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading polars_lts_cpu-1.31.0-cp39-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (34.7 MB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m34.7/34.7 MB\u001b[0m \u001b[31m36.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading python_dotenv-1.1.1-py3-none-any.whl (20 kB)\n","Downloading rich_click-1.8.9-py3-none-any.whl (36 kB)\n","Downloading botocore-1.39.5-py3-none-any.whl (13.8 MB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m13.8/13.8 MB\u001b[0m \u001b[31m81.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading colormath2-3.0.3-py3-none-any.whl (40 kB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m40.7/40.7 kB\u001b[0m \u001b[31m2.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading humanfriendly-10.0-py2.py3-none-any.whl (86 kB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m86.8/86.8 kB\u001b[0m \u001b[31m4.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading jmespath-1.0.1-py3-none-any.whl (20 kB)\n","Downloading s3transfer-0.13.0-py3-none-any.whl (85 kB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m85.2/85.2 kB\u001b[0m \u001b[31m5.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hInstalling collected packages: kaleido, python-dotenv, polars-lts-cpu, jmespath, humanfriendly, colormath2, spectra, coloredlogs, botocore, s3transfer, rich-click, boto3, multiqc\n","Successfully installed boto3-1.39.5 botocore-1.39.5 coloredlogs-15.0.1 colormath2-3.0.3 humanfriendly-10.0 jmespath-1.0.1 kaleido-0.2.1 multiqc-1.30 polars-lts-cpu-1.31.0 python-dotenv-1.1.1 rich-click-1.8.9 s3transfer-0.13.0 spectra-0.1.0\n"]}],"source":["! pip install multiqc"]},{"cell_type":"code","execution_count":8,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":16626,"status":"ok","timestamp":1752629443077,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"},"user_tz":240},"id":"C0tmIpFtRMM9","outputId":"4dc534f0-3c5f-460b-a412-825034681349"},"outputs":[{"output_type":"stream","name":"stdout","text":["\n","\u001b[91m///\u001b[0m \u001b]8;id=11701;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ 🔍 \u001b[2mv1.30\u001b[0m\n","\n","\u001b[34m       file_search\u001b[0m | Search path: /content/drive/MyDrive/Lab_share/Lab_data/class_466/clean.trimgalore.RNA-Seq\n","\u001b[2K         \u001b[34msearching\u001b[0m | \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m30/30\u001b[0m  \n","\u001b[?25h\u001b[34m          cutadapt\u001b[0m | Found 12 reports\n","\u001b[34m     write_results\u001b[0m | Data        : multiqc_data\n","\u001b[34m     write_results\u001b[0m | Report      : multiqc_report.html\n","\u001b[34m           multiqc\u001b[0m | MultiQC complete\n"]}],"source":["!multiqc drive/MyDrive/Lab_share/Lab_data/class_466/clean.trimgalore.RNA-Seq/"]},{"cell_type":"code","source":["! ls -al"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"hwm1k7dZ3UHa","executionInfo":{"status":"ok","timestamp":1752629597679,"user_tz":240,"elapsed":136,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"}},"outputId":"c404325d-56e5-47da-e98b-215f8d474eaf"},"execution_count":11,"outputs":[{"output_type":"stream","name":"stdout","text":["total 4752\n","drwxr-xr-x 1 root root    4096 Jul 16 01:30 .\n","drwxr-xr-x 1 root root    4096 Jul 15 23:38 ..\n","drwxr-xr-x 4 root root    4096 Jul 14 13:37 .config\n","drwx------ 5 root root    4096 Jul 15 23:40 drive\n","drwxr-xr-x 2 root root    4096 Jul 16 01:30 multiqc_data\n","-rw-r--r-- 1 root root 4839639 Jul 16 01:30 multiqc_report.html\n","drwxr-xr-x 1 root root    4096 Jul 14 13:37 sample_data\n"]}]},{"cell_type":"code","execution_count":16,"metadata":{"executionInfo":{"elapsed":1693,"status":"ok","timestamp":1752629985663,"user":{"displayName":"Bioinfo Scholarship Committee","userId":"16207537275623265301"},"user_tz":240},"id":"dFaQ8YgYhz-f","colab":{"base_uri":"https://localhost:8080/","height":1000,"output_embedded_package_id":"1BISlQI7MXeqn4wOJhrYIIXG4Un0bQpx1"},"outputId":"6ab74080-71de-40c4-d1b1-d9ffae28865a"},"outputs":[{"output_type":"display_data","data":{"text/plain":"Output hidden; open in https://colab.research.google.com to view."},"metadata":{}}],"source":["from IPython.display import HTML\n","import os\n","\n","html_file_name = '/content/multiqc_report.html' # Replace with your HTML file\n","html_file_path = os.path.join(os.getcwd(), html_file_name)\n","\n","if not os.path.exists(html_file_path):\n","    print(f\"Error: {html_file_path} not found.\")\n","else:\n","    with open(html_file_path, 'r', encoding='utf-8') as f:\n","        html_content = f.read()\n","\n","    display(HTML(html_content))"]}],"metadata":{"colab":{"provenance":[],"authorship_tag":"ABX9TyOLrT9GGnLsJOfhxmIwDdmd"},"kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"name":"python"}},"nbformat":4,"nbformat_minor":0}